### PR TITLE
Fixes some errors

### PR DESCRIPTION
**Pull request by Soapy7261**
And does more checks!

### CODE REVIEW

File: cogs/socials.py [hunk 1]
```diff
@@ -257,22 +257,26 @@ async def gay(self, ctx, user=None, border=False, server_avatar=True):
...
+                    botmsg = message.content.split("\n")[1] #Same thing but remove the "Paw" part
+                except IndexError:
+                    continue #I guess it wasn't a gpt request like I thought
+                if not usermessage.startswith("**Prompt:**") and not botmsg.startswith("**Paw:**"):
+                    continue
                 if botmsg == "Generating...":
                     continue
-                gpthistory.append({"role": "user", "content": usermessage, "name": message.author.display_name})
-                gpthistory.append({"role": "assistant", "content": botmsg})
+                gpthistory.append({"role": "user", "content": usermessage, "name": message.author.display_name[12:]})
```
Comment (ToothyDev) [inline cogs/socials.py]: What is the [12:] for?

Comment (Soapy7261) [reply cogs/socials.py]: Oops

File: cogs/socials.py [hunk 2]
```diff
@@ -257,22 +257,26 @@ async def gay(self, ctx, user=None, border=False, server_avatar=True):
...
         gpthistory = [{"role": "system", "content": f"{data.gaslight} The user's name is {ctx.author.display_name}. Do not use the user's full name, use their call name derived from their full name."}]
         for message in messages:
             if message.content is None:
                 continue
             if message.author == self.bot.user:
-                #Get the first line of the message
-                usermessage = message.content.split("\n")[0][12:] #Get the first line of the message and remove the "Prompt" part
-                botmsg = message.content.split("\n")[1][9:] #Same thing but remove the "Paw" part
+                try:
+                    usermessage = message.content.split("\n")[0] #Get the first line of the message and remove the "Prompt" part
```
Comment (ToothyDev) [inline cogs/socials.py:270]: How does this remove the "Prompt" part exactly?

Comment (Soapy7261) [reply cogs/socials.py:270]: Line 278

**Review by ToothyDev — APPROVED** [review]

**Pull request merged**